Document data transfer during Contact merge (#759)
mautic/user-documentation: docs/contacts/manage_contacts.rst

.. vale on

If you have 2 Contacts in the Mautic database who are physically one person, you can merge them with the Merge feature.

#. Click the drop down box arrow in the top right hand corner of the Contact detail.
#. Select the Merge item. A modal box shows up.
#. Search for the Contact you want to merge into the current Contact. The select box updates as you search.

> #. Select the right Contact and hit the **Merge** button.

Reviewer: @promptless-for-oss I think it'd be great to add screenshots here for visuality. Add descriptive alt text following accessibility best practice. Fix the grammar in the suggestion if necessary.

```suggestion
#. Click the drop down box arrow in the top right hand corner of the Contact detail.
#. Select the Merge item.

   |
   
   [IMAGE: contact_merge_button.png]
   
   |
   
#. Once a modal box shows up, search for the Contact you want to merge into the current Contact. The select box updates as you search.
#. Select the right Contact and hit the **Merge** button.

   |

   [IMAGE: merge_contacts.png]
```
---

**Images:**

contact_merge_button.png

<img width="2079" height="900" alt="Image" src="https://github.com/user-attachments/assets/86f5ec09-5ba4-4851-88e6-39062e5343ab" />

merge_contacts.png

<img width="1120" height="631" alt="Image" src="https://github.com/user-attachments/assets/cff78b7f-3857-4a0f-8d1d-d057adf49842" />

.. vale on

If you have 2 Contacts in the Mautic database who are physically one person, you can merge them with the Merge feature.

1. Click the drop down box arrow in the top right hand corner of the Contact detail,

2. Select the Merge item, a modal box shows up.

3. Search for the Contact you want to merge into the current Contact. The select box updates as you search.

> 4. Select the right Contact and hit the **Merge** button.

Reviewer: ```suggestion
#. Click the drop down box arrow in the top right hand corner of the Contact detail.
#. Select the Merge item. A modal box shows up.
#. Search for the Contact you want to merge into the current Contact. The select box updates as you search.
#. Select the right Contact and hit the **Merge** button.
```

@promptless-for-oss please address the suggestions.

When you merge Contacts, the surviving Contact receives the following data:

* **Field data** - Values from the merged Contact fill empty fields on the surviving Contact.
* **Owner** - If the surviving Contact doesn't have an owner, it inherits the merged Contact's owner.
* **Points** - Points from both Contacts combine.
* **Tags** - Tags from both Contacts combine.

> * **Company associations** - Company associations from the merged Contact transfer to the surviving Contact. If both Contacts have a primary Company, the surviving Contact keeps its primary. If only the merged Contact has a primary Company, the surviving Contact inherits that primary designation.

Author: Added Company associations bullet point because PR #16247 fixes a bug where company associations were silently lost during Contact merges. The fix adds a `mergeCompanies()` step to `ContactMerger::merge()` that transfers company associations while preserving the primary company designation.

Source: https://github.com/mautic/mautic/pull/16247

.. vale on

If you have 2 Contacts in the Mautic database who are physically one person, you can merge them with the Merge feature.

#. Click the drop-down box arrow in the top right-hand corner of the Contact detail.
#. Select the **Merge** item. A modal box shows up.

   .. image:: images/contact-merge-button.png
      :align: center
      :alt: Merge option highlighted in the Contact actions drop-down menu on the Contact detail.

#. Search for the Contact you want to merge into the current Contact. The select box updates as you search.
#. Select the right Contact and click the **Merge** button.

   .. image:: images/merge-contacts.png
      :align: center

>       :alt: Merge Contact modal with a search field, a Choose the Contact to merge with select box, and the Merge button.

Reviewer: @promptless-for-oss there are a few things:

- Replace the `contact-merge-button.png` with below image and change the name to `contact_merge_option.png`:

  <img width="2079" height="900" alt="Image" src="https://github.com/user-attachments/assets/0a505ab5-3c1b-4d3c-9b04-7dd4116344eb" />

- Change dash to underscore for all image names for consistency. For example: `contact-merge-option.png` become `contact_merge_option.png`.
- Address the suggestion below:

```suggestion
#. Click the drop-down box arrow in the top right-hand corner of the Contact detail.
#. Select the **Merge** item. A modal box shows up.

   |

   .. image:: images/contact_merge_option.png
      :align: center
      :alt: Merge option highlighted in the Contact actions drop-down menu on the Contact detail.
      
   | 

#. Search for the Contact you want to merge into the current Contact. The select box updates as you search.
#. Select the right Contact and click the **Merge** button.

   |

   .. image:: images/merge_contacts.png
      :align: center
      :alt: Merge Contact modal with a search field, a Choose the Contact to merge with select box, and the Merge button.
      
   |
```

.. vale on

> If you have 2 Contacts in the Mautic database who are physically one person, you can merge them with the Merge feature.

Reviewer: ```suggestion
If you have two Contacts in the Mautic database who are physically one person, you can merge them with the Merge feature.
```

#. Click the drop-down box arrow in the top right-hand corner of the Contact detail.
#. Select the **Merge** item. A modal box shows up.

   |

   .. image:: images/contact_merge_option.png
      :align: center
      :alt: Merge option highlighted in the Contact actions drop-down menu on the Contact detail.

   |

> #. Search for the Contact you want to merge into the current Contact. The select box updates as you search.

Reviewer: I'm thinking to:

- move the 'a modal shows up' to the next step, so we can be align with the visual.
- replace 'modal' with 'pop-up window' or 'dialog box', whichever is better in this context, so general user can understand it.

```suggestion
#. Select the **Merge** option.

   |

   .. image:: images/contact_merge_option.png
      :align: center
      :alt: Merge option highlighted in the Contact actions drop-down menu on the Contact detail.

   |

#. Once the pop-up window appears, search for the Contact you want to merge into the current Contact. The select box updates as you search.
```

> #. Select the right Contact and click the **Merge** button.

Reviewer: I think we can clarify and simplify this.

```suggestion
#. Click on the Contact you want to keep. This contact remains after merging.
#. On the Contact detail, click the dropdown arrow in the top right-hand corner.
#. Select the **Merge** option.

   |

   .. image:: images/contact_merge_option.png
      :align: center
      :alt: Merge option highlighted in the Contact actions drop-down menu on the Contact detail.

   |

#. Once the pop-up window appears, type to search for the Contact you want to merge, then open the dropdown menu to select it. Mautic deletes this duplicate Contact after merging.
#. Click the **Merge** button.
```

   |

   .. image:: images/merge_contacts.png
      :align: center
      :alt: Merge Contact pop-up window with a search field, a Choose the Contact to merge with select box, and the Merge button.

   |

When you merge Contacts, the surviving Contact receives the following data:

* **Field data** - Values from the merged Contact fill empty fields on the surviving Contact.
* **Owner** - If the surviving Contact doesn't have an owner, it inherits the merged Contact's owner.
* **Points** - Points from both Contacts combine.
* **Tags** - Tags from both Contacts combine.

> * **Company associations** - Company associations from the merged Contact transfer to the surviving Contact. If both Contacts have a primary Company, the surviving Contact keeps its primary. If only the merged Contact has a primary Company, the surviving Contact inherits that primary designation.

Reviewer: @promptless-for-oss After took a closer look at this, I'm not sure about 'surviving' Contact. It feels too technical.

Would 'primary' or 'main' Contact make a better term for general user without taking the original intention?

If you have 2 Contacts in the Mautic database who are physically one person, you can merge them with the Merge feature.

#. Click the drop-down box arrow in the top right-hand corner of the Contact detail.
#. Select the **Merge** item. A modal box shows up.

   .. image:: images/contact-merge-button.png
      :align: center

>       :alt: Merge option highlighted in the Contact actions drop-down menu on the Contact detail page.

Reviewer: Address the Vale warning: [Mautic.FeatureList] Is this referring to a Mautic feature? If so, use 'Landing Page' instead of 'page'.

#. Search for the Contact you want to merge into the current Contact. The select box updates as you search.
#. Select the right Contact and click the **Merge** button.

   .. image:: images/merge-contacts.png
      :align: center

>       :alt: Merge Contact modal with a search field, a Choose the contact to merge with select box, and the Merge button.

Reviewer: Address the Vale warning: [Mautic.FeatureList] Is this referring to a Mautic feature? If so, use 'Contact' instead of 'contact'.
